File: go/ad-hoc/1486.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	for {
		// Read n, m, and p using bufio.Scanner
		scanner.Scan()
		n, _ := strconv.Atoi(scanner.Text())
		scanner.Scan()
		m, _ := strconv.Atoi(scanner.Text())
		scanner.Scan()
		p, _ := strconv.Atoi(scanner.Text())

		if n == 0 && m == 0 && p == 0 {
			break
		}

		// Initialize a 2D array for the vetor
		vetor := make([][]int, m+1)
		for i := 1; i <= m; i++ {
			vetor[i] = make([]int, n+1)
		}

		// Read vetor values directly into the array
		for i := 1; i <= m; i++ {
			for j := 1; j <= n; j++ {
				scanner.Scan()
				val, _ := strconv.Atoi(scanner.Text())
				vetor[i][j] = val
			}
		}

		resp := 0
		for i := 1; i <= n; i++ {
			jaforam := 0
			for j := 1; j <= m; j++ {
				if vetor[j][i] == 1 {
					jaforam++
				} else {
					if jaforam >= p {
						resp++
					}
					jaforam = 0
				}
			}
			if jaforam >= p {
				resp++
			}
		}

		fmt.Printf("%d\n", resp)
	}
}

File: python3/ad-hoc/1486.py
while True:
    n, m, p = map(int, input().split())
    if n == 0 and m == 0 and p == 0:
        break

    vetor = [[0] * (n + 1) for _ in range(m + 1)]
    for i in range(1, m + 1):
        vetor[i] = [0] + list(map(int, input().split()))

    resp = 0
    for i in range(1, n + 1):
        jaforam = 0
        for j in range(1, m + 1):
            if vetor[j][i] == 1:
                jaforam += 1
            else:
                if jaforam >= p:
                    resp += 1
                jaforam = 0
        if jaforam >= p:
            resp += 1

    print(resp)
